### ISSUE TITLE
NYT10: test set same as validation set?

### ISSUE THREAD

**Issue by fenchri** [thunlp/OpenNRE#267]
Hello there,

Thank you for making this tool open source! I used the corresponding scripts to download the NYT10 corpus and found that the test and validation data are exactly the same.
Is this some kind of error or it is done on purpose?

Thank you in advance!

**Comment by gaotianyu1350** [comment]
Hi~ The original NYT10 dataset does not have a validation set, and for the consistency of the API I just make the val set the same as the test set.

**Comment by fenchri** [comment]
Thanks for your response! 
I noticed however that the model selects the best parameters based on the AUC on the validation set, which is the same as the test set. Isn't it wrong to select the best model on the test set?

**Comment by gaotianyu1350** [comment]
Yep. That's kind of problematic. However, to keep consistent with some previous publications, I still keep the validation set the same as the test set. A recommended way, however, is to split your own validation set from the training set.

Ps. In our provided nyt10 dataset, we have a validation set split from the training set before, but it causes some confusion because it is not the same as the original nyt10 dataset.

**Comment by fenchri** [comment]
Indeed I have seen the previous split, this is why I noticed this. 
By the way, do you provide any processing scripts in case someone wants to pre-process the original NYT data (http://iesl.cs.umass.edu/riedel/ecml/) from scratch? 
Thanks!

**Comment by gaotianyu1350** [comment]
Yep. You can find the original data + preprocessing codes in the `tensorflow` branch [readme](https://github.com/thunlp/OpenNRE/tree/tensorflow).

**Comment by fenchri** [comment]
Great! Thanks a lot :)
